platform/security/security-oidc-realm/src/main/java/org/codice/ddf/security/oidc/realm/OidcRealm.java

```diff
@@ -83,29 +81,15 @@ protected AuthenticationInfo doGetAuthenticationInfo(AuthenticationToken authent
     OidcCredentials credentials = (OidcCredentials) oidcAuthenticationToken.getCredentials();
     OIDCProviderMetadata oidcProviderMetadata =
         oidcHandlerConfiguration.getOidcConfiguration().findProviderMetadata();
-    OidcTokenValidator oidcTokenValidator =
-        new OidcTokenValidator(
-            oidcHandlerConfiguration.getOidcConfiguration(), oidcProviderMetadata);
     WebContext webContext = (WebContext) oidcAuthenticationToken.getContext();
 
-    oidcTokenValidator.validateAccessToken(credentials.getAccessToken(), credentials.getIdToken());
-    if (credentials.getIdToken() != null) {
-      oidcTokenValidator.validateIdTokens(credentials.getIdToken(), webContext);
-    } else {
-      try {
-        OidcAuthenticator authenticator =
-            new CustomOidcAuthenticator(
-                oidcHandlerConfiguration.getOidcConfiguration(),
-                oidcHandlerConfiguration.getOidcClient(),
-                oidcProviderMetadata);
-        authenticator.validate(credentials, webContext);
-      } catch (TechnicalException e) {
-        LOGGER.debug(
-            "Problem fetching id token with credentials ({}) and web context ({}).",
-            credentials,
-            webContext);
-      }
-    }
+    OidcCredentialsResolver oidcCredentialsResolver =
```

Comment:
📘 `OidcCredentialsResolver` is now responsible for resolving the id token and validating all tokens.